Meteo/src/engeto/company/com/AllMeasurements.java

```diff
@@ -0,0 +1,158 @@
+                        Integer.parseInt(separatedData[2]),
+                        Double.parseDouble(separatedData[3]),
+                        Double.parseDouble(separatedData[4]),
+                        Double.parseDouble(separatedData[5]),
+                        Double.parseDouble(separatedData[6]),
+                        Double.parseDouble(separatedData[7]),
+                        Double.parseDouble(separatedData[8]),
+                        Double.parseDouble(separatedData[9]));
+
+                importedMeasurements.add(newMeasurement);
+            }
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+    public List<SingleMeasurement> getImportedMeasurements() {
+        return importedMeasurements;
+    }
+
+    public double getAvgTemp() {
+        double avg_temps = 0.0;
+        for (SingleMeasurement sm : importedMeasurements) {
+            avg_temps += sm.getAvg_temp();
+        }
+        return Math.round((avg_temps /= importedMeasurements.size())*10)/10.0;
+    }
+
+    public SingleMeasurement getExtremeTempMeasurement(String minMax) {
+        SingleMeasurement measurement = new SingleMeasurement();
+
+        if (minMax.equals("max")) {
+            importedMeasurements.sort(new SortByMaxTemp());
+            measurement = importedMeasurements.get(importedMeasurements.size()-1);
+        } else if (minMax.equals("min")) {
+            importedMeasurements.sort(new SortByMinTemp());
+            measurement = importedMeasurements.get(0);
+        }
+        return measurement;
+    }
+
+    public int getAllWindyDays() {
+        int result = 0;
+        for (SingleMeasurement m : importedMeasurements) {
+            if (m.getWindSpeed() >= 4.2) {
+                result++;
+            }
+        }
+        return result;
+    }
+
+    public int getAllCalmDays() {
+        int result = 0;
+        for (SingleMeasurement m : importedMeasurements) {
+            if (m.getWindSpeed() <= 1.8) {
+                result++;
+            }
+        }
+        return result;
+    }
+
+    public double getPrecipitationPerTenDays(int whichDecade) {
+
+        double result = 0.0;
+        int dayNumber = 0;
+        int startPos;
+
+        switch(whichDecade){
+            case 1:
+                startPos = 0;
+                dayNumber = 10;
+                break;
+            case 2:
+                startPos = 10;
+                dayNumber = 20;
+                break;
+            case 3:
+                startPos = 20;
+                dayNumber = importedMeasurements.size();
+                break;
+            default:
+                return result;
+        }
+
+        Collections.sort(importedMeasurements);
+
+        for (int i = startPos; i < dayNumber; i++){
+            SingleMeasurement m = importedMeasurements.get(i);
+            result += m.getPrecipitation();
+        }
+        return Math.round(result*10)/10.0;
+    }
+
+    public void printAsterisks(){
```

Comment:
metodu bych dal mimo tuhle třídu (výpis by měla zajišťovat jiná třída, zavání to porušením prvního solid pravidla)